bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ThreadFactoryBuilder.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.common;
+
+import static java.util.concurrent.Executors.defaultThreadFactory;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A builder for {@link ThreadFactory} instances. This builder is intended to be use for creating thread factories to be
+ * used, e.g., when creating {@link Executor}s via the {@link Executors} utility methods.
+ * <p>
+ * The built {@link ThreadFactory} uses a wrapped {@link ThreadFactory} to create threads (defaulting to
+ * {@link Executors#defaultThreadFactory()}, and then overwrites thread properties as indicated in the build process.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@NonNullByDefault
+public class ThreadFactoryBuilder {
+
+    @Nullable
+    private ThreadFactory wrappedThreadFactory;
```

Comment:
Sorry, I missed that comment before:
I don't think we have any rule here, but I would prefer to add the nullness annotation in front of the type instead of the visibility modifier.
`wrappedThreadFactory` is a nullable `ThreadFactory` and its controlloing access is marked private.

So, please use:
```
private @Nullable ThreadFactory wrappedThreadFactory
```

@triller-telekom Should this added to the nullness documentation?

Reply:
Thanks for that suggestion, looks much better with type annotation directly in front of type.

Reply:
Just checked https://www.eclipse.org/smarthome/documentation/development/conventions.html#null-annotations - on that page it is actually done the other way round (though with `NonNullByDefault` and not `Nullable` - but I guess those should be treated likewise).

@triller-telekom What do you think?

Reply:
@triller-telekom IMHO it should be changed to:

```
private @NonNullByDefault({}) MyService injectedService;
```

IIRC this has already be discussed that we annotate the type and so the annotation should be in front of the type intead of the access modifier. Do you remember?

Reply:
@maggu2810 I remember our discussion, but somehow it didn't make it in the documentation so far :(

I have created a PR to fix this: https://github.com/eclipse/smarthome/pull/5839